### PR TITLE
Threepid validity checks on msisdns should not be dependent on 'threepid_behaviour_email'.

### PR DESCRIPTION
**Pull request by neilisfragile**
A stab at fixing https://github.com/matrix-org/synapse/issues/6103 - got stuck trying to test it

### CODE REVIEW

File: synapse/handlers/auth.py [hunk 1]
```diff
@@ -443,43 +443,48 @@ def _check_threepid(self, medium, authdict, **kwargs):
...
-                threepid_creds["client_secret"],
-                sid=threepid_creds["sid"],
-                validated=True,
-            )
+                raise SynapseError(
+                    400, "SMS delegation is not enabled on this homeserver"
+                )
+        elif medium == "email":
+            if self.hs.config.threepid_behaviour_email == ThreepidBehaviour.REMOTE:
+                if medium == "email":
```
Comment (jryans) [inline synapse/handlers/auth.py]: Already checked this two lines up, so should be able to remove this one.

File: synapse/handlers/auth.py [hunk 2]
```diff
@@ -443,43 +443,48 @@ def _check_threepid(self, medium, authdict, **kwargs):
...
-                # Valid threepid returned, delete from the db
-                yield self.store.delete_threepid_session(threepid_creds["sid"])
+                if row:
+                    # Valid threepid returned, delete from the db
+                    yield self.store.delete_threepid_session(threepid_creds["sid"])
+            else:
+                raise SynapseError(400, "Email is not enabled on this homeserver")
         else:
-            raise SynapseError(
-                400, "Password resets are not enabled on this homeserver"
```
Comment (jryans) [inline synapse/handlers/auth.py]: Thanks for fixing this error! (This method is also used for registration as well, so password reset here was confusing.)

**Comment by richvdh** [comment]
I'm afraid I've refactored all this code in #6105... I'm going to close this PR and rewrite it based on the new branch.

**Comment by richvdh** [comment]
scratch that, i'm going to merge them

**Pull request merged**